pkg/ingester/wal.go

```diff
@@ -215,6 +215,19 @@ func (w *walWrapper) performCheckpoint() (err error) {
 		return err
 	}
 
+	if lastCh == lastSegment {
```

Comment:
Should we check `&& lastCh != -1` to cover the case we have no checkpoints and no segments? In general, I've the feeling that `performCheckpoint()` doesn't handle the case `lastSegment == -1`: what am I missing?

Reply:
I think there should be at-least 1 segment when the WAL is started. But I will add the check just in case.